File: remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} A
 * @return {number}
 */
var removeDuplicates = function(A) {

    if (A.length <= 1) return A.length;

    var pre;
    for (var i = 0; i < A.length;) {
        if (pre === A[i]) {
            A.splice(i, 1);
        } else {
            pre = A[i];
            i ++;
        }
    }

    return A.length;
};

var eq = require('assert').equal;

var t = [1,2,3,4,4,5,5,5,6];
eq(removeDuplicates(t), 6);
eq(removeDuplicates([]), 0);
eq(removeDuplicates([1,1,1,1]), 1);

File: python/remove_duplicates_from_sorted_array.py
class Solution:
    # @param a list of integers
    # @return an integer
    def removeDuplicates(self, A):
        # 1 1 2 2 2 2 3 4 5
        i = 0
        val = None
        while i < len(A):
            if val == A[i]:
                # A.pop(i)
                A[i-1:i+1] = [A[i-1]]
            else:
                val = A[i]
                i += 1

        print A
        return len(A)

s = Solution()

assert s.removeDuplicates([1,1,2,2,3,3,4,5,6]) == 6
assert s.removeDuplicates([1,1]) == 1
assert s.removeDuplicates([]) == 0
